x-pack/plugins/maps/public/components/layer_addpanel/flyout_footer/view.js

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyoutFooter,
+  EuiButtonEmpty,
+  EuiButton,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export const FlyoutFooter = (
+  { onClick, disableButton, buttonText, closeFlyout, hasLayerSelected, isLoading }
+) => {
+
+  const nextButtonText = buttonText || 'Add layer';
+  const nextButton = (
+    <EuiButton
+      disabled={!hasLayerSelected || disableButton}
+      isLoading={hasLayerSelected && isLoading}
```

Comment:
This is not working and `isLoading` is not getting displayed. 

The test cases below are using a document source that contains the military bases that were imported previously.

Here is the view in your branch. Notice how there is no feedback in the `Add layer` button
<img width="400" alt="Screen Shot 2019-05-24 at 9 24 31 AM" src="https://user-images.githubusercontent.com/373691/58339642-46404a00-7e07-11e9-8c2e-ea1fe4685615.png">

And here is the view in master
<img width="300" alt="Screen Shot 2019-05-24 at 9 34 51 AM" src="https://user-images.githubusercontent.com/373691/58339669-522c0c00-7e07-11e9-9411-893042c9039a.png">

Reply:
I think this is fallout from `addLayer` waiting for syncing data. The selected layer is not getting set until the data has loaded

Reply:
Ah, nice catch

Reply:
Actually, this one was interesting. The issue was already fixed upstream for adding layers like in your examples, so that should work fine now. For import, the correct boolean vals are coming through to the component, but the `json.parse` operation in the file upload plugin blocks the thread and clobbers the progress indicator. It's actually a bigger issue that needs to be addressed separately